src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -179,6 +179,10 @@ $TYPEDFIELDS
     rounds::Int = -1
     """maximum number of attempts to make per round (`-1` for infinite)"""
     attempts::Int = -1
+    """function or index of the slot to choose an available free slot in node A"""
+    chooseA::Union{Int,<:Function} = minimum
+    """function or index of the slot to choose an available free slot in node B"""
+    chooseB::Union{Int,<:Function} = minimum
```

Comment:
I am thinking through comparisons of what is done in `SwapperProt` -- Just making sure this will not lead to a lot of confusion for someone who wants to use both.

I think you are right to not permit nodeA/nodeB to be wildcards, they are not like SwapperProt.nodeL/SwapperProt.nodeH, they are more like SwapperProt.node.

However chooseA sounds too much like SwapperProt.chooseL, but chooseA is for slots in a node while chooseL is for a node in a list of nodes.

Maybe we can call it `chooseAslot` and `chooseBslot`?

Reply:
maybe

```
    """function `Vector{Int}->Union{Nothing,Int}` or an integer slot number, specifying the slot to take among available free slots in node A"""
    chooseA::Union{Int,<:Function} = minimum
    """specifying the slot to take among available free slots in node A"""
    chooseB::Union{Int,<:Function} = minimum
```